### ISSUE TITLE
Remove static usage of "tabs" permissions

### ISSUE THREAD

**Issue by dellagustin** [podStation/podStation#54]
Follow up of #26 

# Regarding Tabs Permission

https://developer.chrome.com/extensions/tabs

**Why do we use it:** Tabs is necessary so that we can bring a podstation tab into focus, if it is already open: https://github.com/podStation/podStation/blob/master/extension/background/podstation_bg.js

> The majority of the chrome.tabs API can be used without declaring any permission. However, the "tabs" permission is required in order to populate the url, pendingUrl, title, and favIconUrl properties of Tab.

But we need to query for open tabs of podStation.  
The documentation of the `query` function states the following regarding the `url` property of the `queryInfo` parameter:

> Match tabs against one or more URL patterns. Fragment identifiers are not matched. This property is ignored if the extension does not have the "tabs" permission.

I tested removing the `"tabs"` permission and changing the `url` of the `queryInfo` object into `chrome-extension://${chrome.runtime.id}/podstation.html` with the ropes the rule would not apply in for tabs belonging to the extension, no luck.

- [x] I can try using the "activeTab" permission - see https://github.com/podStation/podStation/issues/54#issuecomment-578546311
- [x] I can try to ask the permission dynamically and see what happens. - OBSOLETE
- [x] I should also raise a feature request to chromium so that I can query for my own tabs. - done - https://bugs.chromium.org/p/chromium/issues/detail?id=1045813
- [x] produce a reduced example that shows the feature request
- [x] I should also check the specifications for Browser Extensions - and raise this use case - https://github.com/browserext/browserext/issues/59

## Current solution proposal

- Use the "activeTab" permission instead of "tabs"
- Use method getViews and getCurrent, see https://github.com/podStation/podStation/issues/54#issuecomment-578663722

## References 

* See the references of #26
* https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions
* https://browserext.github.io/browserext/
* https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Working_with_the_Tabs_API

**Comment by dellagustin** [comment]
I tested with the "activeTab" permission and it seems to generate the same effect while not showing the same permission warning.

**Comment by dellagustin** [comment]
There was a suggestion at https://bugs.chromium.org/p/chromium/issues/detail?id=1045813 to try to following piece of code:

```js
Promise.all(chrome.extension.getViews({type: 'tab'}).map(view =>
  new Promise(resolve =>
    view.chrome.tabs.getCurrent(tab => {
      tab.url = view.location.href;
      resolve(tab);
    }))
)).then(ownTabs => {
  console.log(ownTabs);
});
```

I am skeptical due to `view.chrome.tabs.getCurrent`, but I want to experiment with it.

**Comment by dellagustin** [comment]
Reading now the documentation of the suggested APIs, I think the suggestion I mention on https://github.com/podStation/podStation/issues/54#issuecomment-578663722 will actually work.
It is even better then the "activeTab" permission, as no additional permission seem to be necessary.

**Comment by dellagustin** [comment]
Implemented with #60
